source/ide/wiki/wiki_methods.bas

```diff
@@ -1068,6 +1075,51 @@ FUNCTION wikiSafeName$ (page$) 'create a unique name for both case sensitive & i
     wikiSafeName$ = page$ + "_" + ext$
 END FUNCTION
 
+$UNSTABLE:HTTP
+FUNCTION wikiDLPage$ (url$, timeout!)
+'--- set default result & avoid side effects ---
+wikiDLPage$ = ""
+wik$ = url$: tio! = timeout!
+'--- open client ---
+retry:
+ch& = _OPENCLIENT(wik$)
+IF Help_Recaching < 2 THEN 'avoid messages for 'qb64pe -u' (build time update)
+    IF ch& = 0 AND LCASE$(LEFT$(wik$, 8)) = "https://" THEN
+        IF _MESSAGEBOX("QB64-PE Help", "Can't make secure connection (https:) to Wiki, shall the IDE use unsecure (http:) instead?", "yesno", "warning" ) = 1 THEN
+            IF _MESSAGEBOX("QB64-PE Help", "Do you wanna save your choice permanently for the future?", "yesno", "question" ) = 1 THEN
+                wikiBaseAddress$ = "http://" + MID$(wikiBaseAddress$, 9)
+                WriteConfigSetting generalSettingsSection$, "WikiBaseAddress", wikiBaseAddress$
+            END IF
+            wik$ = "http://" + MID$(wik$, 9): GOTO retry
+        END IF
+    END IF
+END IF
+IF ch& = 0 THEN EXIT FUNCTION
+'--- wait for response ---
+res$ = "": st! = TIMER
+DO
+    _DELAY 0.05
+    GET ch&, , rec$
+    IF LEN(rec$) > 0 THEN st! = TIMER
+    res$ = res$ + rec$
+    IF _STATUSCODE(ch&) = 200 THEN
+        le& = LOF(ch&)
+        IF le& > -1 THEN
+            IF LEN(res$) = le& THEN

```

Comment:
This works fine but a little unnecessary, checking `EOF(ch&)` always works so you could skip the `LOF(ch&)` logic and just always do the `EOF(ch&)` check to end the loop.